src/Microsoft.DotNet.ProjectModel/Workspace/ProjectContextEntry.cs

```diff
@@ -0,0 +1,207 @@
+                {
+                    LastLockFileWriteTime = DateTime.MinValue;
+                }
+                else
+                {
+                    var lockFilePath = Path.Combine(value.ProjectFile.ProjectDirectory, LockFile.FileName);
+                    LastLockFileWriteTime = File.Exists(lockFilePath) ? File.GetLastWriteTime(lockFilePath) :
+                                                                        DateTime.MinValue;
+                }
+
+                Diagnostics.Clear();
+                Dependencies.Clear();
+                ProjectReferences.Clear();
+                FileReferences.Clear();
+                SourceFiles.Clear();
+                HasDependencyResolved = false;
+                _context = value;
+            }
+        }
+
+        public DateTime LastLockFileWriteTime { get; private set; }
+
+        public bool HasChanged
+        {
+            get
+            {
+                if (ProjectContext == null)
+                {
+                    return true;
+                }
+
+                var lockFilePath = Path.Combine(ProjectContext.ProjectFile.ProjectDirectory,
+                                                LockFile.FileName);
+
+                return LastLockFileWriteTime < File.GetLastWriteTime(lockFilePath);
+            }
+        }
+
+        public bool HasDependencyResolved { get; private set; }
+
+        public List<DiagnosticMessage> Diagnostics { get; } = new List<DiagnosticMessage>();
+
+        public List<DependencyDescription> Dependencies { get; } = new List<DependencyDescription>();
+
+        public List<ProjectReferenceInfo> ProjectReferences { get; } = new List<ProjectReferenceInfo>();
+
+        public List<string> FileReferences { get; } = new List<string>();
+
+        public List<string> SourceFiles { get; } = new List<string>();
+
+        public void Reset()
+        {
+            _context = null;
+            HasDependencyResolved = false;
+
+            Diagnostics.Clear();
+            Dependencies.Clear();
+            ProjectReferences.Clear();
+            FileReferences.Clear();
+            SourceFiles.Clear();
+
+            LastLockFileWriteTime = DateTime.MinValue;
+        }
+
+        public void GetDependencyInfo(string configuration)
+        {
+            Diagnostics.Clear();
+            Dependencies.Clear();
+            ProjectReferences.Clear();
+            FileReferences.Clear();
+            SourceFiles.Clear();
+
+            Diagnostics.AddRange(ProjectContext.LibraryManager.GetAllDiagnostics());
+            SourceFiles.AddRange(ProjectContext.ProjectFile.Files.SourceFiles);
+
+            var libraries = ProjectContext.LibraryManager.GetLibraries();
+            var librariesLookup = libraries.ToLookup(lib => lib.Identity.Name);
+            var diagnosticSources = Diagnostics.ToLookup(diagnostic => diagnostic.Source);
+            ProjectDescription mainProject = null;
+
+            foreach (var library in libraries)
```

Comment:
This loop isn't really required to be separate from the other loop. Take a look at how these APIs are used in the rest of the CLI. .e.g

https://github.com/dotnet/cli/blob/master/src/Microsoft.DotNet.Tools.Publish/Program.cs#L132